tests/Unit/Domain/CoordinateMaps/Test_ProductMaps.cpp

```diff
@@ -56,290 +56,317 @@ void test_product_two_maps_fail() {
-  CHECK(jac_xi.get(0, 0) == jacobian_00);
-
-  CHECK(jac_A.get(0, 1) == 0.0);
-  CHECK(jac_B.get(0, 1) == 0.0);
-  CHECK(jac_xi.get(0, 1) == 0.0);
-
-  CHECK(jac_A.get(0, 2) == 0.0);
-  CHECK(jac_B.get(0, 2) == 0.0);
-  CHECK(jac_xi.get(0, 2) == 0.0);
-
-  CHECK(jac_A.get(1, 0) == 0.0);
-  CHECK(jac_B.get(1, 0) == 0.0);
-  CHECK(jac_xi.get(1, 0) == 0.0);
-
-  CHECK(jac_A.get(1, 1) == jacobian_11);
-  CHECK(jac_B.get(1, 1) == jacobian_11);
-  CHECK(jac_xi.get(1, 1) == jacobian_11);
-
-  CHECK(jac_A.get(1, 2) == 0.0);
-  CHECK(jac_B.get(1, 2) == 0.0);
-  CHECK(jac_xi.get(1, 2) == 0.0);
-
-  CHECK(jac_A.get(2, 0) == 0.0);
-  CHECK(jac_B.get(2, 0) == 0.0);
-  CHECK(jac_xi.get(2, 0) == 0.0);
-
-  CHECK(jac_A.get(2, 1) == 0.0);
-  CHECK(jac_B.get(2, 1) == 0.0);
-  CHECK(jac_xi.get(2, 1) == 0.0);
-
-  CHECK(jac_A.get(2, 2) == jacobian_22);
-  CHECK(jac_B.get(2, 2) == jacobian_22);
-  CHECK(jac_xi.get(2, 2) == jacobian_22);
-
-  // Check Jacobians for DataVectors
-  const Mesh<3> mesh{8, Spectral::Basis::Legendre,
-                     Spectral::Quadrature::GaussLobatto};
-  const auto tensor_logical_coords = logical_coordinates(mesh);
-  const std::array<DataVector, 3> logical_coords{
-      {tensor_logical_coords.get(0), tensor_logical_coords.get(1),
-       tensor_logical_coords.get(2)}};
-  const auto volume_inv_jac = affine_map_xyz.inv_jacobian(logical_coords);
-  const auto volume_jac = affine_map_xyz.jacobian(logical_coords);
-  for (size_t i = 0; i < 3; ++i) {
-    for (size_t j = 0; j < 3; ++j) {
-      if (i == j) {
-        CHECK(volume_inv_jac.get(i, j) ==
-              DataVector(logical_coords[0].size(),
-                         i == 0 ? inv_jacobian_00
-                                : i == 1 ? inv_jacobian_11 : inv_jacobian_22));
-        CHECK(volume_jac.get(i, j) ==
-              DataVector(
-                  logical_coords[0].size(),
-                  i == 0 ? jacobian_00 : i == 1 ? jacobian_11 : jacobian_22));
-      } else {
-        CHECK(volume_inv_jac.get(i, j) ==
-              DataVector(logical_coords[0].size(), 0.0));
-        CHECK(volume_jac.get(i, j) ==
-              DataVector(logical_coords[0].size(), 0.0));
+    CHECK_FALSE(affine_map_xy != affine_map_xy);
+    test_serialization(affine_map_xy);
+
+    test_coordinate_map_argument_types(affine_map_xy, point_xi);
+    test_product_two_maps_fail();
+
+    check_if_map_is_identity(
+        CoordinateMaps::ProductOf2Maps<CoordinateMaps::Affine,
+                                       CoordinateMaps::Affine>{
+            CoordinateMaps::Affine{-1.0, 1.0, -1.0, 1.0},
+            CoordinateMaps::Affine{-1.0, 1.0, -1.0, 1.0}});
+    CHECK(not CoordinateMaps::ProductOf2Maps<CoordinateMaps::Affine,
+                                             CoordinateMaps::Affine>{
+        CoordinateMaps::Affine{-1.0, 2.0, -1.0, 1.0},
+        CoordinateMaps::Affine{-1.0, 1.0, -3.0, 1.0}}
+                  .is_identity());
+  }
+  SECTION("Product of 3 Maps") {
```

Comment:
I think it would be much cleaner if you separated these sections into functions. I personally prefer that style because it keeps functions small and reduces how much scope can spill. For example, you can put two functions `void test_product_2_maps()` and `void product_3_maps()` into an anonymous namespace and call them both from a single `SPECTRE_TEST_CASE`